components/Editor/Editor_Side/AttendeeFormEditor.tsx

```diff
@@ -0,0 +1,203 @@
+'use client';
+import React, { useEffect, useState } from 'react';
+import Button from '@/components/Button';
+import saveTemplate, { getAllTemplateIds } from '@/utils/save-template';
+import queryForTemplate from '@/utils/queryTemplate';
+import EditorColumn from './EditorRightColumn';
+
+interface AttendeeInfo {
+  id: number;
+  label: string;
+  accessKey: string;
+  type: string;
+  placeholder: string;
+}
+
+interface PageProps {
+  userId: string;
+  eventId: string;
+}
+//needs
+export default function NewAttendeeFormEditor({
+  params,
+  attendeeInfoList,
+  setAttendeeInfoList,
+  setBackgroundColor,
+  setTextColor
+}: {
+  params: PageProps;
+  attendeeInfoList: AttendeeInfo[];
+  setAttendeeInfoList: React.Dispatch<React.SetStateAction<AttendeeInfo[]>>;
+  setBackgroundColor: React.Dispatch<React.SetStateAction<string>>;
+  setTextColor: React.Dispatch<React.SetStateAction<string>>;
+}) {
+  const capitalizeFirstLetter = (str: string): string =>
+    `${str.charAt(0).toUpperCase()}${str.slice(1)}`;
+
+  let defaultList: AttendeeInfo[] = [];
+  // This data is the previously saved template.
+  useEffect(() => {
+    const fetchTemplate = async () => {
+      try {
+        console.log('trying to queryForTemplate');
+        console.log(
+          'userID: ' + params.userId + ' ;eventID: ' + params.eventId
+        );
+        const templateData = await queryForTemplate(
+          params.userId,
+          params.eventId
+        );
+
+        // if there is templateData and event_id isnt the only key in that templateData
+        if (templateData && Object.keys(templateData)[0] != 'event_id') {
+          console.log('templateData', templateData);
+          // Rendering main form
+          let i = 0;
+          defaultList = Object.entries(templateData)
+            .filter(
+              ([key, value]) =>
+                key !== 'event_id' &&
```

Comment:
This might be due to mismatching keys naming convention from our previous version, for newly created events this point onwards, this wont be the case I believe. For older events, I'll migrate them to follow the new convention.